test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsFileParsingTests/CredentialsItemTests.cs

```diff
@@ -280,6 +280,33 @@ public void CredentialsItem_Update_WhenOriginIsMachineWide_Throws()
             }
         }
 
+        [Fact]
+        public void CredentialsItem_Update_WhenOriginIsReadOnly_Throws()
+        {
+            // Arrange
+            using (var mockBaseDirectory = TestDirectory.Create())
+            {
+                var origin = new SettingsFile(mockBaseDirectory, fileName: Settings.DefaultSettingsFileName, isMachineWide: false, isReadOnly: true);
+
+                var xelement = new XElement("name",
+                                new XElement("add", new XAttribute("key", "Username"), new XAttribute("value", "user")),
+                                new XElement("add", new XAttribute("key", "Password"), new XAttribute("value", "pass")));
+
+                var credentials = new CredentialsItem(xelement, origin);
+
+                // Act
+                var ex = Record.Exception(() =>
+                    credentials.Update(new CredentialsItem("name", "user", "notpass", isPasswordClearText: true, validAuthenticationTypes: null)));
+
+                // Assert
+                ex.Should().NotBeNull();
+                ex.Should().BeOfType<InvalidOperationException>();
+                ex.Message.Should().Be("Unable to update setting since it is in an uneditable config file.");
```

Comment:
```suggestion
                ex.Message.Should().Be(Resources.CannotUpdateReadOnlyConfig);
```

Consider adding InternalsVisibleTo this test project to NuGet.Configuration

Reply:
internals visible to is already there.

Change it Resources.*